src/main/java/it/gov/pagopa/payhub/activities/service/exportflow/debtposition/BaseExportFlowFileService.java

```diff
@@ -0,0 +1,180 @@
+        this.csvService = csvService;
+        this.csvRowDtoClass = csvRowDtoClass;
+        this.fileArchiverService = fileArchiverService;
+        this.workingDirectory = workingDirectory;
+        this.relativeFileFolder = relativeFileFolder;
+        this.fileNamePrefix = fileNamePrefix;
+    }
+
+    public ExportFlowFileResult executeExport(Long exportFileId) {
+        int[] pageNumber = {0};
+        long[] exportedRows = {0};
+
+        E exportFileRecord = findExportFileRecord(exportFileId);
+        ExportFileStatus exportStatus = getExportStatus(exportFileRecord);
+
+        if (exportStatus.equals(ExportFileStatus.PROCESSING)){
+            Long organizationId = getOrganizationId(exportFileRecord);
+            String profile = getFlowFileVersion(exportFileRecord);
+            Path csvFilePath = workingDirectory.resolve(String.valueOf(organizationId)).resolve(relativeFileFolder).resolve(getExportFileName(exportFileId));
+            F exportFilter = getExportFilter(exportFileRecord);
+
+            try {
+                csvService.createCsv(csvFilePath, csvRowDtoClass, () ->{
+                            List<C> csvRows = retrieveAndMapPage(exportFileRecord, exportFilter, pageNumber[0]++);
+                            exportedRows[0] += csvRows.size();
+                            return csvRows;
+                        }
+                        , profile);
+            } catch (IOException e) {
+                throw new IllegalStateException("Error writing to CSV file: " + e.getMessage(), e);
+            }
+
+            Path zipFilePath = createZipArchive(csvFilePath);
+
+            return ExportFlowFileResult.builder()
+                    .fileName(zipFilePath.getFileName().toString())
+                    .filePath(zipFilePath.toString())
+                    .exportedRows(exportedRows[0])
+                    .build();
+
+        }else {
+            throw new InvalidExportStatusException("The requested ExportFile (%s) has an invalid status %s".formatted(exportFileId, exportStatus));
+        }
+    }
+
+    /**
+     * Creates a ZIP archive for the specified CSV file.
+     * This method compresses the provided CSV file into a ZIP archive using the {@link FileArchiverService#compressAndArchive(List, Path, Path)} method.
+     *
+     * @param csvFilePath The path to the CSV file to be compressed.
+     * @return The path to the created ZIP archive.
+     * @throws IllegalStateException If an error occurs during compression and archiving due to filesystem or permission issues.
+     */
+    private Path createZipArchive(Path csvFilePath) {
+        try {
+            Path tmpZipFilePath = csvFilePath.getParent()
+                    .resolve(Utilities.replaceFileExtension(csvFilePath.getFileName().toString(), ".zip"));
+            fileArchiverService.compressAndArchive(List.of(csvFilePath), tmpZipFilePath, workingDirectory);
+            return tmpZipFilePath;
```

Comment:
this is still the working directory filePath, you need to return the folder relative to organization folder